### PR TITLE
Refactor ensure_pam_wheel_group_empty rule

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

Refactor `ensure_pam_wheel_group_empty` rule ir order to make it simpler and more efficient.
This PR includes the following improvements:
- Simplify the OVAL assessment logic with two simple tests
  - The same object is used in both tests so no unnecessary calls are made
  - The two tests make it clear which parts of the assessment are failing
  - The report no longer includes unnecessary objects (all empty groups)
- Optimize test scenarios by removing unnecessary commands, while keeping more compatible commands
- Simplify the Bash remediation by removing unnecessary commands
- Make the Ansible remediation more efficient

More details about the changes in commit descriptions.

#### Rationale:

- Make it easier to interpret the report, specially when the rule fails in a system.
- Simpler and more efficient rules
- Fixes https://issues.redhat.com/browse/RHEL-1905

#### Review Hints:

I would recommend to check the commits and their respective descriptions one by one to see the context of the changes.
Technically, executing automatus should be enough to test the rule:
Bash
`./tests/automatus.py rule --libvirt qemu:///session rhel8 --datastream build/ssg-rhel8-ds.xml --dontclean --remediate-using bash ensure_pam_wheel_group_empty`

Ansible
`./tests/automatus.py rule --libvirt qemu:///session rhel8 --datastream build/ssg-rhel8-ds.xml --dontclean --remediate-using ansible ensure_pam_wheel_group_empty`

This is an example of how is the report after this PR:
Screenshot from 2023-10-11 09-26-09

### CODE REVIEW

**Comment by marcusburghardt** [comment]
Hi @teacup-on-rockingchair , I see the `ensure_pam_wheel_group_empty` rule is failing for SLE15 because the `groupmems` is not found. This command was there even before this PR so I assume this rule remediation never worked in SLE15. Could you take a look and give some hints on how to make it work in SLE15 too, please? We can update this PR or you can open a new PR once this is merged. What do you prefer?

File: linux_os/guide/system/accounts/accounts-restrictions/root_logins/ensure_pam_wheel_group_empty/rule.yml [hunk 1]
```diff
@@ -2,18 +2,17 @@ documentation_complete: true
...
+title: 'Ensure the Group Used by pam_wheel.so Module Exists on System and is Empty'
 
 description: |-
-    Ensure that the group {{{ xccdf_value("var_pam_wheel_group_for_su") }}}
-    referenced by the pam_wheel  <tt>group</tt> parameter exists and has no
-    members. This ensures that no user can run commands with altered
-    privileges through the <tt>su</tt> command.
+    Ensure that the <tt>{{{ xccdf_value("var_pam_wheel_group_for_su") }}}</tt> group referenced by
+    the <tt>pam_wheel.so</tt> <tt>group</tt> option exists and has no members. This ensures
+    that no user can run commands with altered privileges through the <tt>su</tt> command.
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-restrictions/root_logins/ensure_pam_wheel_group_empty/rule.yml:10]: I still find this slightly confusing. My confusion comes from the fact that here we talk about `pam_wheel.so` module option, but this rule's OVAL and OCIL don't check any PAM configuration, it only checks the existence of a group in `/etc/group`. From this context the group name might seem arbitrary. Apparently, the PAM configuration is handled by rule `use_pam_wheel_group_for_su` and both rules are selected together to provide an experience. But If I imagine that I'm a user that suddenly sees this rule failing I think i might not be aware of this connection with the other rule. I'm not sure how to reword this description to reflect that.

Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-restrictions/root_logins/ensure_pam_wheel_group_empty/rule.yml]: @jan-cerny could you take a look in the last two commits, please?

**Review by jan-cerny — APPROVED** [review]
I like this PR very much. I have reviewed the code. I have run test scenarios locally. I have viewed all HTML reports. I can say that the warning and the layout helps a lot to understand the fail in some situations.

Unfortunately, in the test scenario `ensure_pam_wheel_group_empty-no_group.fail.sh` we again hit https://github.com/OpenSCAP/openscap/issues/1507.

**Comment by jan-cerny** [comment]
@marcusburghardt A quick google search hints that they might intentionally remove `groupmems` from SUSE https://forums.opensuse.org/t/groupmems-command-not-found/153854/2. But, I think that you can bypass the nonexistence of the `groupmems` command by first deleting the group and then adding it. Another option would be to list the member of the group and remove each of them from the group.

**Comment by marcusburghardt** [comment]
> @marcusburghardt A quick google search hints that they might intentionally remove `groupmems` from SUSE https://forums.opensuse.org/t/groupmems-command-not-found/153854/2. But, I think that you can bypass the nonexistence of the `groupmems` command by first deleting the group and then adding it. Another option would be to list the member of the group and remove each of them from the group.

Thanks for the reference @jan-cerny . I will include a Jinja2 conditional in this Bash remediation so we can move forward.

**Comment by teacup-on-rockingchair** [comment]
> Hi @teacup-on-rockingchair , I see the `ensure_pam_wheel_group_empty` rule is failing for SLE15 because the `groupmems` is not found. This command was there even before this PR so I assume this rule remediation never worked in SLE15. Could you take a look and give some hints on how to make it work in SLE15 too, please? We can update this PR or you can open a new PR once this is merged. What do you prefer?

Thanks @marcusburghardt and @jan-cerny for the heads up. I can confirm it never worked and we enabled the bash remediation for our platform by mistake. I will open an issue to handle that in a separate context.

**Comment by marcusburghardt** [comment]
> > Hi @teacup-on-rockingchair , I see the `ensure_pam_wheel_group_empty` rule is failing for SLE15 because the `groupmems` is not found. This command was there even before this PR so I assume this rule remediation never worked in SLE15. Could you take a look and give some hints on how to make it work in SLE15 too, please? We can update this PR or you can open a new PR once this is merged. What do you prefer?
> 
> Thanks @marcusburghardt and @jan-cerny for the heads up. I can confirm it never worked and we enabled the bash remediation for our platform by mistake. I will open an issue to handle that in a separate context.

Thanks for the update and for opening the https://github.com/ComplianceAsCode/content/issues/11203 @teacup-on-rockingchair . @jan-cerny , I in this case the error is not a regression but something that was already there in the past and we already have an Upstream issue to fix it (in a separate PR). I believe we can waive the error and merge the PR.

**Comment by jan-cerny** [comment]
@marcusburghardt I agree with your proposal.

**Pull request merged**